internal/artifact/service/artifact.go

```diff
@@ -41,11 +47,45 @@ func (A *ArtifactService) GetArtifactList(pageNum int64, rows int64) (list Artif
 	}
 
 	list.TotalData = alist.TotalData
-	list.Artifacts = []ArtifactItem{}
+	list.Artifacts = []artifactModel.Artifact{}
 
 	for _, a := range alist.Artifacts {
-		list.Artifacts = append(list.Artifacts, ArtifactItem{Name: a.Name})
+		list.Artifacts = append(list.Artifacts, artifactModel.Artifact{Name: a.Name})
+	}
+
+	return
+}
+
+// SubmitPackage submit package
+func (A *ArtifactService) SubmitPackage(tarball string) (job SubmissionJob, err error) {
+	submittedJob := artifactModel.Submission{
+		Timestamp: time.Now(),
+	}
+
+	submittedJob.TaskUUID = generateSubmissionUUID(submittedJob.Timestamp)
+
+	err = A.repo.PutTarballToFile(&tarball, submittedJob.TaskUUID)
+	if err != nil {
+		return job, errors.New("Can't store tarball " + err.Error())
+	}
+
+	err = A.repo.ExtractSubmittedTarball(submittedJob.TaskUUID)
+	if err != nil {
+		return job, errors.New("Can't extract tarball " + err.Error())
 	}
 
+	// verify the package signature
+	isVerified, err := A.repo.VerifyArtifact(submittedJob.TaskUUID)
+	if err != nil || !isVerified {
+		return job, errors.New("Can't verify tarball")
+	}
+
+	// TODO : we haven't delete the tarball here
```

Comment:
For now, the upload has been fixed by:
- Uploading the tarball to /v1/submission-upload to get the blob id. They're accessible on https://irgsh.blankonlinux.or.id/submissions/.
- Send submission metadata that included the tarball blob id to get the pipeline ID

Did you plan that this v2 API is providing submission in one strike? I'm thinking about the metadata is included in the tarball.